modules/grpc/otel/otel-dest.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2023 Attila Szakacs
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "otel-dest.hpp"
+
+#define get_DestDriver(s) (((OtelDestDriver *) s)->cpp)
+
+using namespace syslogng::grpc::otel;
+
+/* C++ Implementations */
+
+DestDriver::DestDriver(OtelDestDriver *s)
+  : super(s)
+{
+}
+
+const char *
+DestDriver::generate_persist_name()
+{
+  static char persist_name[1024];
+
+  if (super->super.super.super.super.persist_name)
+    g_snprintf(persist_name, sizeof(persist_name), "opentelemetry.%s",
+               super->super.super.super.super.persist_name);
+  else
+    g_snprintf(persist_name, sizeof(persist_name), "opentelemetry");
+
+  return persist_name;
+}
+
+const char *
+DestDriver::format_stats_instance()
+{
+  static char stats_instance[1024];
+
+  if (super->super.super.super.super.persist_name)
+    g_snprintf(stats_instance, sizeof(stats_instance), "opentelemetry,%s",
+               super->super.super.super.super.persist_name);
+  else
+    g_snprintf(stats_instance, sizeof(stats_instance), "opentelemetry");
```

Comment:
This needs to be adjusted if #4551 goes in first.

Reply:
Thanks, rebased and adjusted.